tests/saitests/py3/sai_qos_tests.py

```diff
@@ -226,9 +279,76 @@ def fill_leakout_plus_one(test_case, src_port_id, dst_port_id, pkt, queue, asic_
                 print("fill_leakout_plus_one: Success, sent %d packets, queue occupancy bytes rose from %d to %d" % (
                     packet_i + 1, queue_counters_base[queue], queue_counters[queue]), file=sys.stderr)
                 return True
+        else:
+            raise RuntimeError("Couldnot fill leakout:src_id:{}, dst_id:{},"
+                " pkt:{}, queue:{}, asic_type:{}".format(
+                    src_port_id, dst_port_id, "{}".format(pkt.__repr__()[0:200]),
+                    queue, asic_type))
+
     return False
 
 
+def get_multiple_flows_udp(dp, dst_mac, dst_id, dst_ip, src_vlan, dscp, ecn,
+        ttl, pkt_len, src_details, packets_per_port=1):
+    '''
+        Returns a dict of format:
+        src_id : [list of (pkt, exp_pkt) pairs that go to the given dst_id]
+    '''
+
+    def get_rx_port_udp(dp, src_port_id, pkt, exp_pkt):
+        # 3 attempts to account for arp/neighbor entry.
+        result = None
+        for i in range(3):
+            send_packet(dp, src_port_id, pkt, 1)
+
+            result = dp.dataplane.poll(
+                device_number=0, exp_pkt=exp_pkt, timeout=3)
+            if isinstance(result, dp.dataplane.PollFailure):
+                # Try again.
+                continue
+
+        if isinstance(result, dp.dataplane.PollFailure):
+            dp.fail("Expected packet was not received."
+                "Received on port:{} {}".format(
+                result.port, result.format()))
+
+        return result.port
+
+    udp_port_gen = get_udp_port()
+    all_pkts = {}
+    for src_tuple in src_details:
```

Comment:
better to unpack "src tuple" to get readable,
for example:
for src_port_id, src_port_ip, src_port_mac in src_details: